packages/mocktail/lib/src/_register_matcher.dart

```diff
@@ -16,46 +16,111 @@ Map<Type, Object?> _createInitialFallbackValues() {
   createValue<String>('42');
   createValue<Object>('42');
   createValue<dynamic>('42');
-  createValue<Map<String, dynamic>>(const <String, dynamic>{});
-  createValue<Map<String, Object>>(const <String, Object>{});
-  createValue<Map<String, Object>>(const <String, Object>{});
-  createValue<Map<String?, dynamic>>(const <String, dynamic>{});
-  createValue<Map<String?, Object>>(const <String, Object>{});
-  createValue<Map<String?, Object?>>(const <String, Object>{});
-  createValue<Set<int>>(const <int>{});
-  createValue<Set<int?>>(const <int?>{});
-  createValue<Set<double>>(const <double>{});
-  createValue<Set<double?>>(const <double?>{});
-  createValue<Set<num>>(const <num>{});
-  createValue<Set<num?>>(const <num?>{});
-  createValue<Set<String>>(const <String>{});
-  createValue<Set<String?>>(const <String?>{});
-  createValue<Set<bool>>(const <bool>{});
-  createValue<Set<bool?>>(const <bool?>{});
-  createValue<Set<Object>>(const <Object>{});
-  createValue<Set<Object?>>(const <Object?>{});
-  createValue<Set<dynamic>>(const <dynamic>{});
-  createValue<List<int>>(const <int>[]);
-  createValue<List<int?>>(const <int>[]);
-  createValue<List<double>>(const <double>[]);
-  createValue<List<double?>>(const <double?>[]);
-  createValue<List<num>>(const <num>[]);
-  createValue<List<num?>>(const <num?>[]);
-  createValue<List<String>>(const <String>[]);
-  createValue<List<String?>>(const <String?>[]);
-  createValue<List<Object>>(const <Object>[]);
-  createValue<List<Object?>>(const <Object?>[]);
-  createValue<List<bool>>(const <bool>[]);
-  createValue<List<bool?>>(const <bool?>[]);
-  createValue<List<dynamic>>(const <dynamic>[]);
 
   return result;
 }
 
+Never _fallbackCallback([
+  Object? a1,
+  Object? a2,
+  Object? a3,
+  Object? a4,
+  Object? a5,
+  Object? a6,
+  Object? a7,
+  Object? a8,
+  Object? a9,
+  Object? a10,
+  Object? a11,
+  Object? a12,
+  Object? a13,
+  Object? a14,
+  Object? a15,
+  Object? a16,
+  Object? a17,
+  Object? a18,
+  Object? a19,
+  Object? a20,
+]) {
+  throw UnsupportedError(
+    '''
+A test tried to call mockito\'s internal dummy callback.
+This dummy callback is only meant to be passed around, but never called.''',
+  );
+}
+
+List<Object?> _genericFallbackValues = [
+  const <Never>[],
+  const <Never, Never>{},
+  const <Never>{},
+  _fallbackCallback,
+];
+
 final _fallbackValues = _createInitialFallbackValues();
 
+T _getFallbackValue<T>() {
+  bool isFunction<T>() {
+    return T.toString().contains('=>');
```

Comment:
This is kind of a hack. I don't know if there is a better way to check if `T` is a `Function`.

Reply:
This fails when we have a generic type which has a type parameter such as `void Function()`. I do think that it is useful to distinguish between what kind of fallback is missing to provide a useful error message, but to do that correctly we're going to have to write a parser for `runtimeType`.

Reply:
I created a parser for `runtimeType` (even though that feels like a really bad idea). https://github.com/miDeb/runtime_type_parser if you're interested.

Reply:
@miDeb while this is a very creative solution, I don't think it makes sense to include as part of the package as `runtimeType` can be overridden and is not a reliable way to make this determination.

Reply:
Yeah that's what I thought as well. Let's just create a general error message including all the different ways to register fallback types.

Reply:
Alright, I updated the error message to not use runtimeType in the code samples. I also reworded it a bit.

Reply:
> I don't think it makes sense to include as part of the package as runtimeType can be overridden

_Just_ to clarify, this isn't actually a problem because we're not calling `runtimeType` on an instance, but we get the Type as a type parameter which has to be correct. Still not a good idea to parse it though :smile: .